### PR TITLE
Update .ghal.rules.json for v3

### PR DESCRIPTION
**Pull request by Thraka**
You can't really look at the diff here because the structure changed heavily. I ported the existing ruleset. I need to merge before I can deploy the tool update.

The new tool does not support the old v1 and v2 schemas anymore.

### CODE REVIEW

**Review by mairaw — APPROVED** [review]
:shipit:

File: .ghal.rules.json [hunk 1]
```diff
@@ -1,113 +1,275 @@
...
-              "(?i).*docs\/fsharp.*": ":books: Area - F# Guide",
-              "(?i).*docs\/machine-learning.*": ":books: Area - ML.NET Guide",
-              "(?i).*docs\/spark.*": ":books: Area - .NET for Apache Spark Guide",
-              "(?i).*docs\/standard.*": ":books: Area - .NET Guide",
-              "(?i).*docs\/standard\/security.*": ":card_file_box: Technology - Security",
-              "(?i).*docs\/standard\/standard\/design-guidelines\/.*": ":book: guide - Framework Design Guidelines",
-              "(?i).*docs\/visual-basic.*": ":books: Area - Visual Basic Guide"
+    "opened": {
+      "processor-default": {
+        "milestone-set": "![month]"
```
Comment (Thraka) [inline .ghal.rules.json]: @mairaw @BillWagner I think we talked about having the milestone automatically assigned, yes? This config line does that when a PR is opened. It looks for a milestone named `MMMM yyyy` which translates to (for this month) `October 2019`.

Comment (mairaw) [reply .ghal.rules.json]: For PRs?

File: .ghal.rules.json [hunk 2]
```diff
@@ -1,113 +1,275 @@
...
+      },
+      "processor-conditional": {
+        "type": "query",
+        "value": "PullRequest.base.ref == 'master'",
+        "processors": {
+          "processor-files": {
```
Comment (Thraka) [inline .ghal.rules.json]: @mairaw for reference, this section uses a new conditional processor that lets you run a query over the json schema of the PR, Issue, or Comment. So in this case, when a PR is opened it checks if the PR targets master and then proceeds to the nested processor which is the changed files detection which adds labels. 

This should solve your problem with opening PRs that target live.

File: .ghal.rules.json [hunk 3]
```diff
@@ -1,113 +1,275 @@
...
+            "changedfile": {
+              "(?i).*docs\/architecture.*": {
```
Comment (Thraka) [inline .ghal.rules.json]: @mairaw for reference, this section uses a new conditional processor that lets you run a query over the json schema of the PR, Issue, or Comment. So in this case, when a PR is opened it checks if the PR targets master and then proceeds to the nested processor which is the changed files detection which adds labels. 

This should solve your problem with opening PRs that target live.

File: .ghal.rules.json (hunk 1 above)
Comment (mairaw) [reply .ghal.rules.json]: Then yes, that would be cool

File: .ghal.rules.json (hunk 3 above)
Comment (mairaw) [reply .ghal.rules.json]: Nice!

**Pull request merged**